src/EditorFeatures/CSharp/DecompiledSource/CSharpDecompiledSourceService.cs

```diff
@@ -103,14 +103,26 @@ public static async Task<Document> FormatDocumentAsync(Document document, Syntax
 
             var fullTypeName = new FullTypeName(fullName);
 
-            // Try to decompile; if an exception is thrown the caller will handle it
-            var text = decompiler.DecompileTypeAsString(fullTypeName);
+            try
+            {
+                var text = decompiler.DecompileTypeAsString(fullTypeName);
+
+                text += "#if false // " + CSharpEditorResources.Decompilation_log + Environment.NewLine;
+                text += logger.ToString();
+                text += "#endif" + Environment.NewLine;
 
-            text += "#if false // " + CSharpEditorResources.Decompilation_log + Environment.NewLine;
-            text += logger.ToString();
-            text += "#endif" + Environment.NewLine;
+                return document.WithText(SourceText.From(text));
+            }
+            catch (NotSupportedException)
+            {
+                // IL Spy throws for "Decompiling types that are not part of the main module is not supported."
```

Comment:
Should we just not call them for these in the first place?

Reply:
Yeah, I thought about that, but that just means copying some logic from ILSpy into here, which seems fragile. As per the above, I don't think an NFW when ILSpy can't do something is helping us in any way